### PR TITLE
Update snowflake with permissions

### PR DESCRIPTION
**Pull request by matt-winkler**
## Description & motivation

Customers ask for documentation on the explicit permissions needed for dbt to work with snowflake. Historically, we've pointed them to the "How we configure Snowflake" blog, but that doesn't include the explicit grant statements needed to make it work. This page is intended to be more explicit about what permissions to use as a starting point.

## Checklist
If you added new pages (delete if not applicable):
- [X] The page has been added to `website/sidebars.js`
- [X] The new page has a unique filename

### CODE REVIEW

File: website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md
Comment (dbeatty10) [inline website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md]: Suggesting to capitalize "Snowflake" to match references elsewhere.
```suggestion
**Note:** A crucial part of working with dbt atop Snowflake is ensuring that users (in development environments) and/or service accounts (in deployment to production environments) have the correct permissions to take actions on Snowflake! Here is documentation of the [dbt Labs recommended Snowflake permissions](snowflake-permissions).
```

File: website/docs/reference/snowflake-permissions.md
Comment (dbeatty10) [inline website/docs/reference/snowflake-permissions.md]: Suggesting a newline in the comment since there isn't line wrapping in the browser. See screenshot below for what it looks like currently without this suggestion: 
```suggestion
-- NOTE: warehouse_name, database_name, and role_name are placeholders!
-- Replace as-needed for your organization's naming convention!
```

image

**Comment by matt-winkler** [comment]
Thanks for keeping this up to date @dataders. Anything else needed before merging?

**Review by dbeatty10 — APPROVED** [review]
Thanks for these updates @matt-winkler !

Approving now -- will update to the current branch one more time before merging.

**Pull request merged**
